packages/fx-core/src/component/generator/officeAddin/generator.ts

```diff
@@ -102,12 +103,19 @@ export class OfficeAddinGenerator {
       | "typescript";
     const projectType = inputs[QuestionNames.ProjectType];
     const capability = inputs[QuestionNames.Capabilities];
-    const host: string =
-      projectType === ProjectTypeOptions.outlookAddin().id
-        ? "outlook"
-        : projectType === ProjectTypeOptions.officeAddin().id
-        ? "wxpo" // wxpo - support word, excel, powerpoint, outlook
-        : inputs[QuestionNames.OfficeAddinHost];
+    const inputHost = inputs[QuestionNames.OfficeAddinHost];
+    let host: string = inputHost;
+    if (
+      projectType === ProjectTypeOptions.outlookAddin().id ||
+      (projectType === ProjectTypeOptions.officeXMLAddin().id &&
+        inputHost === OfficeAddinHostOptions.outlook().id)
+    ) {
+      host = "outlook";
+    } else if (capability === CapabilityOptions.officeContentAddin().id) {
+      host = "xp"; // content add-in support excel, powerpoint
+    } else if (capability === "json-taskpane") {
```

Comment:
Just curious that, why just make `content add-in` (but not include `json-taskpane`) avaliable in `CapabilityOptions`?

Reply:
`json-taskpane` exists by default. It is defined in `OfficeAddinProjectConfig` after Huajie's refactor. See it on Create.ts 560 lines.

Reply:
What I means is that why not add `json-taskpane` as a static function in `CapabilityOptions` (file create.ts).

Keep the condition to both use static class or string will look more readbility.

Reply:
I see it. But in my opinion, these part look really confuse since the capability come from static class and dynamic push item.

Consider refactor it later on your demand.

These two condtions use the same right judgement format from static class or string will look better.